src/meltano/core/settings_store.py

```diff
@@ -962,6 +980,7 @@ def get(
         Returns:
             A tuple the got value and an empty dictionary.
         """
+        assert self.session is not None  # noqa: S101
```

Comment:
I'm not thrilled about this, but the dynamic nature of how store managers are instantiated doesn't let us making it a required keyword argument for the DB manager.

Reply:
It's a hacky way of doing type-narrowing. I've tried a different approach in 3526cdf1dc640e3744dcea95db4e586b24e5a39f.

Reply:
I understand that narrowing the type is the goal, but I think an if-statement with a `raise ...` can be used instead. In any case, the property approach seems better.